### ISSUE TITLE
After update to 9.1.5, "Edit" or "Add user to organization" is only available to newly created users

### ISSUE THREAD

**Issue by fliot** [grafana/grafana#55461]
The kubernetes pod is configured to use "grafana/grafana" docker image, then, after redeployment, it migrated by it-self to version 9.1.5 (I don't know what was the earlier version). Migration seems effective, everything is functional, except one tight point.

In the Admin > Users panel, logged as "Grafana admin" (with the small shield icon), I can not anymore modify a user, or assign an organization to existing users.

If I create a new user, for this newly created user, I can modify/edit/assign organization to him, but in any case, not on users that have been created before migration.

Previously created users, seem being "read only" for the administrator.

**Environment**:
- Grafana version: 9.1.5
- Data source type & version: irrelevant
- OS Grafana is installed on: Kubernetes/Docker
- User OS & Browser:  irrelevant
- Grafana plugins: none related to authentication

**Comment by fliot** [comment]
@shyimo , sorry, no I don't know the different affected versions, I just observed it after migration to 9.1.5.
You did observed it on 9.1.4.
Anybody else ?

**Comment by fliot** [comment]
I just update to 9.1.6, and I had the very same lock, but this time, a tiny message was displayed into the user panel !
![image](https://user-images.githubusercontent.com/3476379/191324669-d69b88aa-07c3-4bfb-9c79-3ee51cf172a4.png)

And indeed, for user, that was created in grafana, and never mapped to an third party authentication, I have no issue, and it's only once third party authentication is mapped, that I can not edit any more these users.

1) This is a real bug, even if some part of users information can come from outside world, 
**_as grafana admin_, I need to be able to assign/remove organization, and even disable a user...**

2) I found some quick workaround,
I added SQLITE plugin,
added /var/lib/grafana/grafana.db as SQLITE datasource,
and I edited it with grafana "explore" functionalities.
```
SELECT * FROM user;
SELECT * FROM user_auth;
DELETE FROM user_auth WHERE user_id = 13;  -- where 13 is your read-only user_id
```

It gave me the possibility to edit this user again, from the user panel, even if few minutes later, when the user had logged again, auth mapper mechanism put him in read only again.

My hint is not a real solution, but definitively, a simple workaround.

**Comment by shyimo** [comment]
This is indeed a real bug. i'm not able to change organizations for current users.
As far as we can see, this is true for all 9.1.x versions. 

We did downgrade back to 9.0.9 version in order to get back this specific functionally until this will be resolved

**Comment by mellieA** [comment]
The legacy auth behavior has been that changes were allowed in the UI, but then the next time the user logged in via an external provider their settings would be overwritten by the external authentication configuration.

The new behavior introduced by https://github.com/grafana/grafana/pull/52160 is that the buttons in the UI will disappear so there isn't a false impression that the admin can use the UI to manage the user's role.

There will be a configurable setting that will disable the role_sync behavior and allow an admin to manage the role in the UI - https://github.com/grafana/grafana/pull/55182 

Check out these related PRs, it looks like they should clarify this and give a warning about upcoming changes. The update in 54688 is that this should be implemented in 9.2, but it looks like there is also a backport for 9.1.x coming in a patch release:

[Backport: Revert changes to oauth behavior to v9.1.x](https://github.com/grafana/grafana/pull/55302)
[Auth: Restore legacy behavior and add deprecation notice for empty org role in oauth ](https://github.com/grafana/grafana/pull/55118)
[Auth: Take into account oauth_skip_org_role_update_sync for locking down UI](https://github.com/grafana/grafana/issues/54688)

**Comment by fliot** [comment]
Indeed, duplicate,
Thanks @mellieA
